extensions/wikia/EmbeddableDiscussions/models/DiscussionsThreadModel.class.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+class DiscussionsThreadModel {
+	const DISCUSSIONS_API_BASE = 'https://services.wikia.com/discussion/';
+	const DISCUSSIONS_API_BASE_DEV = 'https://services.wikia-dev.com/discussion/';
+	const DISCUSSIONS_API_SORT_KEY = 'trending';
+	const DISCUSSIONS_API_SORT_DIRECTION = 'descending';
+	const THREAD_CACHE_KEY = "embeddable_discussions_thread";
+	const SORT_TRENDING = 'trending';
+	const SORT_LATEST = 'creation_date';
+	const SORT_TRENDING_LINK = 'trending';
+	const SORT_LATEST_LINK = 'latest';
+
+	const MCACHE_VER = '1.0';
+
+	private $cityId;
+
+	public function __construct( $cityId ) {
+		$this->cityId = $cityId;
+	}
+
+	private function getRequestUrl( $showLatest, $limit ) {
+		global $wgDevelEnvironment;
+
+		$sortKey = $showLatest ? self::SORT_LATEST : self::SORT_TRENDING;
+
+		if ( empty( $wgDevelEnvironment ) ) {
+			return self::DISCUSSIONS_API_BASE . "$this->cityId/threads?sortKey=$sortKey&limit=$limit&viewableOnly=false";
```

Comment:
Created a subtask for this: JPN-631.

If it's ok with you, I'd like to handle this as refactoring to avoid blocking other subtasks.